src/PrestaShopBundle/Twig/SearchFiltersExtension.php

```diff
@@ -0,0 +1,80 @@
+<?php
+/**
+ * 2007-2018 PrestaShop
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to http://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2018 PrestaShop SA
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+namespace PrestaShopBundle\Twig;
+
+use Symfony\Component\HttpFoundation\RequestStack;
+use PrestaShop\PrestaShop\Core\Search\ControllerAction;
+
+/**
+ * Improve information from request, get the real action and controller's names.
+ */
+final class SearchFiltersExtension extends \Twig_Extension
+{
+    /**
+     * @var RequestStack
+     */
+    private $requestStack;
+
+    public function __construct(RequestStack $requestStack)
+    {
+        $this->requestStack = $requestStack;
```

Comment:
Why store requestStack when you use only the current request?

Reply:
Because you are never sure to get a Request => https://github.com/symfony/symfony/blob/master/src/Symfony/Component/HttpFoundation/RequestStack.php#L57

Reply:
So it will raise an exception if you try to use getController or getAction if there's no check if getCurrentRequest sin't null :)

Reply:
i've deleted it :p